src/syntax/Parser.js

```diff
@@ -1412,6 +1429,23 @@ export class Parser {
                               collection, body);
   }
 
+  // The for-on Statement
+  // for  (  { let | var }  identifier  on  expression  )  statement
```

Comment:
Does it make sense? The identifier is assigned the values to be observed. (See also `parseForOfStatement_` in the same file.)